modules/core/src/main/java/org/apache/ignite/internal/IgniteServicesImpl.java

```diff
@@ -75,6 +75,30 @@ public IgniteServicesImpl(GridKernalContext ctx, ClusterGroupAdapter prj, boolea
         return prj;
     }
 
+    /** {@inheritDoc} */
+    @Override public ServiceConfiguration clusterSingletonConfiguration(String name, Service svc) {
+        return ctx.service().serviceConfiguration(name, svc, 1, 1);
+    }
+
+    /** {@inheritDoc} */
+    @Override public ServiceConfiguration nodeSingletonConfiguration(String name, Service svc) {
+        return ctx.service().serviceConfiguration(name, svc, 0, 1);
+    }
+
+    /** {@inheritDoc} */
+    @Override public ServiceConfiguration keyAffinitySingletonConfiguration(String name, Service svc, String cacheName,
```

Comment:
According to codestyle, you should rewrite declaration as
``` 
public ServiceConfiguration keyAffinitySingletonConfiguration(
    String name,
    Service svc,
    String cacheName,
    Object affKey
) {
  ......
```

Reply:
Fixed.